src/util/preemptive-rate-limit.ts

```diff
@@ -0,0 +1,52 @@
+import { createInstallationClient } from "utils/get-github-client-config";
+import { Octokit } from "@octokit/rest";
+import { numberFlag, NumberFlags } from "config/feature-flags";
+import { SQSMessageContext } from "~/src/sqs/sqs.types";
+import { SqsQueue } from "~/src/sqs/sqs";
+import Logger from "bunyan";
+
+// List of queues we want to apply the preemptive rate limiting on
+const TARGETTED_QUEUES = ["backfill"];
+
+// Fetch the rate limit from GitHub API and check if the usages has exceeded the preemptive threshold
+export const preemptiveRateLimitCheck = async (context: SQSMessageContext<any>, sqsQueue: SqsQueue<any>) : Promise<boolean> => {
```

Comment:
It is just personal idea, I find `isRateLimitExceeded()` more readable. Generally for function returning `boolean`, I would let the function name start with `isXXX` or `hasXXX`

Reply:
Hmm, okay, I see this function does more than just checking, it also change the visiblity of the message. 
In that case, function name seems okay. 

/nit
Although I find it a bit hidden, as the `sqsQueue.changeVisibilityTimeout` is both called in this file, and in `src/sqs/sqs.ts` above.

Reply:
haha yeah i went through the same name changing thoughts. In an ideal world I would have it as seperate call to check then have the visibility timeout as a result of the bool, but its a bit of a hassle to force it in there.

this was the tidiest option I could find.